src/main/java/com/owncloud/android/lib/common/utils/Log_OC.java

```diff
@@ -42,51 +42,69 @@
     static class LegacyImpl implements Adapter {
 
         @Override
-        public void i(String tag, String message) {
-            Log.i(tag, message);
-            appendLog(tag + ": " + message);
+        public void i(final String tag, final String message) {
+            if (tag != null && message != null) {
+                Log.i(tag, message);
+                appendLog(tag + ": " + message);
+            }
         }
 
         @Override
-        public void d(String tag, String message) {
-            Log.d(tag, message);
-            appendLog(tag + ": " + message);
+        public void d(final String tag, final String message) {
+            if (tag != null && message != null) {
+                Log.d(tag, message);
+                appendLog(tag + ": " + message);
+            }
         }
 
         @Override
-        public void d(String tag, String message, Exception e) {
-            Log.d(tag, message, e);
-            appendLog(tag + ": " + message + " Exception: " + Arrays.toString(e.getStackTrace()));
+        public void d(final String tag, final String message, Exception e) {
+            if (tag != null && message != null && e != null) {
+                Log.d(tag, message, e);
+                StackTraceElement[] trace = e.getStackTrace();
+                appendLog(tag + ": " + message + " Exception: " + Arrays.toString(trace));
```

Comment:
Is there a reason you separated this? Could also still be inline, or?

Reply:
I'm not sure what you mean my "separation" vs "inline"?

Reply:
The logger implementation in `Log_OC` was extracted so we can use new logger implementation without re-writing modifying world. The application swaps this implementation for it's own `Logger`, but the legacy code (other apps?) can still use the old implementation.

https://github.com/nextcloud/android/pull/4275

Reply:
Previously it was:
Arrays.toString(e.getStackTrace()))

now it is:
StackTraceElement[] trace = e.getStackTrace();
Arrays.toString(trace)